module/Olcs/src/Controller/Factory/Messages/LicenceConversationMessagesControllerFactory.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Olcs\Controller\Factory\Messages;
+
+use Common\Service\Helper\FlashMessengerHelperService;
+use Common\Service\Helper\FormHelperService;
+use Common\Service\Helper\TranslationHelperService;
+use Common\Service\Script\ScriptFactory;
+use Common\Service\Table\TableFactory;
+use Laminas\Navigation\Navigation;
+use Interop\Container\ContainerInterface;
+use Laminas\ServiceManager\FactoryInterface;
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use Laminas\View\HelperPluginManager;
+use Olcs\Controller\Messages\LicenceConversationMessagesController;
+
+class LicenceConversationMessagesControllerFactory implements FactoryInterface
+{
+    /**
+     * @param  ContainerInterface $container
+     * @param  $requestedName
+     * @param  array|null         $options
+     * @return LicenceConversationMessagesController
+     */
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null): LicenceConversationMessagesController
+    {
+        $container = method_exists($container, 'getServiceLocator') ? $container->getServiceLocator() : $container;
+
+        $formHelper = $container->get(FormHelperService::class);
+        assert($formHelper instanceof FormHelperService);
```

Comment:
Do these `assert` function provide value here?

**PHP Docs:**
> Assertions should be used as a debugging feature only. One use case for them is to act as sanity-checks for preconditions that should always be true and that if they aren't upheld this indicates some programming errors. 

https://www.php.net/manual/en/function.assert.php

Reply:
It's something @jerotire has often promoted, I'm following his lead here
I'll leave it out for now, and let him make his case for it another time

Reply:
I think given the fact they can be disabled via a configuration option it's definitely best to avoid them.

Reply:
I've removed asserts from this file, there are others that can be removed but out of scope here
(pretty sure we can convince James to remove them)